src/bootstrap/flags.rs

```diff
@@ -240,6 +245,10 @@ To learn more about a subcommand, run `./x.py <subcommand> -h`",
         opts.optopt("", "rust-profile-use", "use PGO profile for rustc build", "PROFILE");
         opts.optflag("", "llvm-profile-generate", "generate PGO profile with llvm built for rustc");
         opts.optopt("", "llvm-profile-use", "use PGO profile for llvm build", "PROFILE");
+        opts.optflag("A", "", "allow certain clippy lints");
+        opts.optflag("D", "", "deny certain clippy lints");
+        opts.optflag("W", "", "warn about certain clippy lints");
+        opts.optflag("F", "", "forbid certain clippy lints");
```

Comment:
This should be `optopt`, not `optflag`. `optopt` allows passing a value, optflag does not.